src/inference/src/dev/core_impl.cpp

```diff
@@ -936,9 +954,21 @@ ov::SoPtr<ov::ICompiledModel> ov::CoreImpl::compile_model(const std::string& mod
         cacheContent.blobId =
             ov::ModelCache::compute_hash(model_str, weights, create_compile_config(plugin, parsed._config));
         std::unique_ptr<CacheGuardEntry> lock = cacheGuard.get_hash_lock(cacheContent.blobId);
+        
+        std::shared_ptr<ov::Model> model = nullptr;
+        bool is_weightless_caching = is_weightless_caching_enabled(parsed._config);
+
+        if (is_weightless_caching) {
+            model = read_model(model_str, weights);
+            cacheContent.model = model;
+        }
+
         compiled_model =
             load_model_from_cache(cacheContent, plugin, parsed._config, ov::SoPtr<ov::IRemoteContext>{}, [&]() {
-                auto model = read_model(model_str, weights);
+                
+                if (!is_weightless_caching)
```

Comment:
This should be reverted, is fall back path when model cannot be read from cache so it is standard read of model